lib/rules/no-unused-prop-types.js

```diff
@@ -1058,7 +995,7 @@ module.exports = {
       ObjectPattern: function(node) {
         // If the object pattern is a destructured props object in a lifecycle
         // method -- mark it for used props.
-        if (isNodeALifeCycleMethod(node.parent.parent)) {
+        if (astUtil.inConstructor(context) || astUtil.inLifeCycleMethod(context)) {
```

Comment:
also here

Reply:
Same as above.

Fails when removing inConstructor:
* https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/no-unused-prop-types.js#L1228

When removing the `inLifecycleMethod` then 5 tests fail (2 for `componentWillReceiveProps`, 1 for `shouldComponentUpdate` and 2 for `componentDidUpdate`)

I suppose that we can also test https://reactjs.org/docs/react-component.html#componentwillupdate which also has `nextProps` as the first argument.